### ISSUE TITLE
CloudBerry Labs support?

### ISSUE THREAD

**Issue by bigtfromaz** [jlesage/docker-cloudberry-backup#14]
@jlesage I licensed CloudBerry Backup and am running it on the Synology using your image.  However there is a problem.  There is a setting where you can specify the number of threads can start up.  Among other things threads get used to upload files in parallel.  However when running in the container it appears to be uploading 1 file at a time.  It took my initial backup 12 days to complete.  The last 8 were used to upload a large number of small files totaling 30 GB.  

So I opened an issue with CloudBerry Labs and they have pointed out that Alpine Linux is not on their list of supported operating systems.  Would it be difficult to build the container on Ubuntu? Also, can you think of anything that might be restricting the number of threads, or causing their threads to serialize?  

From here down is a copy of the response I received from CloudBerry Labs.  

------------------------------------------------------------------------------

> Hello,

Thanks a lot for your waiting and patience.

As far as we can see, you're actually using the version which is not in our list of supported OS: 

	System type                     : alpine
	System version                  : Alpine Linux v3.9

 ![tmp](https://user-images.githubusercontent.com/7926086/62152623-c4154b80-b2b7-11e9-90ae-e124df5e55cb.png)
Can you please provide more clarity about your setup, so we could consider further investigation? 

We look forward to hearing from you.

**Comment by jlesage** [comment]
I don't think that the fact that CBB is running in a container has anything to do with the issue.  Binaries use shared libraries that come from their own package or from Ubuntu.

It seems that first support level try to filter issues before they reach the second level.  Looking at the OS is an easy way to do it.  Maybe you can refer them to https://github.com/jlesage/docker-cloudberry-backup/issues/3 ;)

**Comment by bigtfromaz** [comment]
A quick update.  CloudBerry Labs decided to go ahead and open a support ticket after I sent them some configuration information taken from the Synology host along with the hardware specs.  

There first move from level 1 was to tell me that the CPU can only run 4 threads.   I correct their miss-perception.  As proof I offered to start the container using the command line instead of the GUI.  I did that adding the --cpuset-cpus 0-3 parameter.

That didn't change the performance.  So I created a special backup, designed to recreate the problem without needing to send 2 TB of data.  I turns out that if I don't encrypt the backup and only use compression, it runs much faster sending 700 files in about 30 seconds.  That's the opposite of my experience.  Compression is usually far more expensive than reversible encryption.

It appears that when encryption is turned on in the backup it single threads the uploads.  CPU demand never exceeded 27% during any test.   I guess it may be possible that there is a less-than-optimal encryption library being chosen in this configuration.

The short answer is that the container seems fine and there is a performance issue in the product.  It'll be interesting to see if they take ownership.

**Comment by bigtfromaz** [comment]
@jlesage OK we have come full circle.  The CloudBerry Labs support tech that was handling my case passed it to another person, ostensibly to someone better equipped to work on the problem.   Silly me, the new person is Level 1, and that person began the conversation by telling me once again that they don't support Alpine.  

In order to remove that excuse I plan to copy your Dockerfile, and base it on Ubuntu.  Do you have any suggestions regarding which version I should use?  Also, is there anything special about the build or will a simple git pull followed by a "docker build" do the trick?

**Comment by jlesage** [comment]
You can use the latest version (`jlesage/baseimage-gui:ubuntu-18.04-v3.5.2`).
And yes, running `docker build -t jlesage/cloudberry-backup .` should be enough.

**Comment by jlesage** [comment]
Did you try to install the application directly on another PC or a VM?  It would be even better if you can reproduce like this.

**Comment by jlesage** [comment]
So the entropy would cause the use of a single thread?
Do they tell how we can determine if the entropy is enough?
